### PR TITLE
Some build.d fixes

### PR DESCRIPTION
**Pull request by marler8997**
Remove source files duplicated between `lexer.a` and the `dmd` executable.  Also add support for relative filename targets.

### CODE REVIEW

**Comment by thewilsonator** [comment]
I would expect a negative line count for deduplication... are you missing something?

**Comment by marler8997** [comment]
The file lists are generated based on iterating through the filesystem, so they filenames aren't duplicated in `build.d`, they are just duplicated in the source list for dmd and the lexer.

Also I added a couple lines to support relative pathnames.

**Comment by thewilsonator** [comment]
OK.

**Review by thewilsonator — APPROVED** [review]

**Review by JinShil — APPROVED** [review]

File: src/build.d [hunk 1]
```diff
@@ -470,11 +470,12 @@ LtargetsLoop:
 
             default:
                 // check this last, target paths should be checked after predefined names
+                const tAbsolute = t.absolutePath.buildNormalizedPath;
                 foreach (dep; DependencyRange(rootDeps.map!(a => a()).array))
                 {
                     foreach (depTarget; dep.targets)
                     {
-                        if (depTarget.endsWith(t))
+                        if (depTarget.endsWith(t) || depTarget.endsWith(tAbsolute))
```
Comment (wilzbach) [inline src/build.d]: `endsWith` is variadic: https://dlang.org/library/std/algorithm/searching/ends_with.html

**Comment by wilzbach** [comment]
> Also I added a couple lines to support relative pathnames.

You used `absolutePath`.

**Comment by marler8997** [comment]
> You used absolutePath.

Yeah, I convert the user's path to an absolute path and compare it to the absolute path of the targets, so that way if either one is relative then they will match.

File: src/build.d (hunk 1 above)
Comment (marler8997) [reply src/build.d]: Incorporated this in https://github.com/dlang/dmd/pull/10203

**Pull request merged**